src/nedap/speced/def/impl/spec_assertion.clj

```diff
@@ -20,15 +20,29 @@
        (catch ~(if clj?
                  `ExceptionInfo
                  'ExceptionInfo) e#
-         (let [spec# (:spec (ex-data e#))
-               inferred-specs# (set (map :spec (extract-specs-from-metadata {:tag ~spec-sym} ~clj?)))]
+         (let [{ex-data-spec#        :spec
+                ex-data-spec-object# :spec-object
+                ex-data-quoted-spec# :quoted-spec
+                :as                  ex-data-object#} (ex-data e#)
+               legacy?# (not (contains? ex-data-object# :spec-object))
+               spec# (if legacy?#
+                       ex-data-spec#
+                       ex-data-spec-object#)
+               inferred-specs# (->> (extract-specs-from-metadata {:tag ~spec-sym} ~clj?)
+                                    (map :spec)
+                                    set)]
 
            ;; rethrow if no spec failure is found
            (when-not spec#
              (throw e#))
 
-           (if (or (= ~spec-sym spec#)
-                   (contains? inferred-specs# spec#))
+           (if (or (if legacy?#
+                     (= ~spec-sym spec#)
+                     (or (= ex-data-quoted-spec# ~spec-sym)
+                         (= ex-data-spec-object# ~spec-sym)))
+                   (contains? inferred-specs# (if legacy?#
+                                                spec#
+                                                ex-data-quoted-spec#)))
              (~reporter {:type :pass, :message ~msg :expected '~spec-sym, :actual nil})
              (~reporter {:type :fail, :message ~msg :expected '~spec-sym, :actual spec#}))
```

Comment:
Is this reporting still correct?

Reply:
Will give it a go soon

Reply:
Good call, improved.

Here's some sample output after it:

![image](https://user-images.githubusercontent.com/1162994/72483436-3eec0780-3801-11ea-99e0-bc7f768d7be0.png)

Reply:
Nice! This makes me think we should eventually add tests for the failure-results as well. We figured out a way to do that in utils.spec: https://github.com/nedap/utils.test/blob/9e3fb51d9f2cd44cb1aeae3b733e23aa4ffb5653/test/unit/nedap/utils/test/api.cljc#L160-L174

Reply:
Mmm, unfortunately I just found out that that test doesn't run correctly in cljs. It boils down to the fact that you can't `with-redefs` a macro there.

Check the cljs compilation output: https://circleci.com/gh/nedap/utils.test/137

Reply:
> It boils down to the fact that you can't with-redefs a macro there.

Even if you could, clojure.test and cljs.test behave pretty differently under the hood.

As discussed privately this test could be `:clj`-only 👍